huaweicloud/services/workspace/resource_huaweicloud_workspace_app_nas_storage.go

```diff
@@ -0,0 +1,290 @@
+		d.Set("created_at", utils.FormatTimeStampRFC3339(utils.ConvertTimeStrToNanoTimestamp(utils.PathSearch("create_time",
+			storageConfig, "").(string))/1000, false)),
+	)
+	if err = mErr.ErrorOrNil(); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func resourceAppNasStorageDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg       = meta.(*config.Config)
+		region    = cfg.GetRegion(d)
+		httpUrl   = "v1/{project_id}/persistent-storages/{storage_id}"
+		storageId = d.Id()
+	)
+
+	client, err := cfg.NewServiceClient("appstream", region)
+	if err != nil {
+		return diag.Errorf("error creating Workspace APP client: %s", err)
+	}
+
+	deletePath := client.Endpoint + httpUrl
+	deletePath = strings.ReplaceAll(deletePath, "{project_id}", client.ProjectID)
+	deletePath = strings.ReplaceAll(deletePath, "{storage_id}", storageId)
+	deleteOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	_, err = client.Request("DELETE", deletePath, &deleteOpt)
+	if err != nil {
+		// Although the deletion result of the main region shows that the interface returns a 200 status code when
+		// deleting a non-existent NAS storage, in order to avoid the possible return of a 404 status code in the
+		// future, the CheckDeleted design is retained here.
+		return common.CheckDeletedDiag(d, err, "NAS Storage of Workspace APP")
+	}
+	return nil
+}
+
+func listAppNasStorages(client *golangsdk.ServiceClient) ([]interface{}, error) {
+	var (
+		httpUrl = "v1/{project_id}/persistent-storages?limit=100"
+		listOpt = golangsdk.RequestOpts{
+			KeepResponseBody: true,
+		}
+		offset = 0
+		result = make([]interface{}, 0)
+	)
+
+	listPath := client.Endpoint + httpUrl
+	listPath = strings.ReplaceAll(listPath, "{project_id}", client.ProjectID)
+
+	for {
+		listPathWithOffset := fmt.Sprintf("%s&offset=%d", listPath, offset)
+		requestResp, err := client.Request("GET", listPathWithOffset, &listOpt)
+		if err != nil {
+			return nil, err
+		}
+		respBody, err := utils.FlattenResponse(requestResp)
+		if err != nil {
+			return nil, err
+		}
+		items := utils.PathSearch("items", respBody, make([]interface{}, 0)).([]interface{})
+		if len(items) < 1 {
+			break
+		}
+		result = append(result, items...)
+		offset += len(items)
+	}
+
+	return result, nil
+}
+
+func resourceAppNasStorageImportState(_ context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	var (
+		cfg         = meta.(*config.Config)
+		region      = cfg.GetRegion(d)
+		storageName = d.Id()
+	)
+
+	client, err := cfg.NewServiceClient("appstream", region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating Workspace APP client: %s", err)
+	}
+
+	storages, err := listAppNasStorages(client)
+	if err != nil {
+		return nil, fmt.Errorf("error getting NAS storage of Workspace APP: %s", err)
+	}
+
+	storageObj := utils.PathSearch(fmt.Sprintf("[?name=='%s']|[0]", storageName), storages, nil)
+
```

Comment:
expression := fmt.Sprintf("[?name=='%s']|[0].id", storageName)
d.SetId(utils.PathSearch(expression , storages, "").(string))

Reply:
sure about that